### PR TITLE
test.py: remove top level conftest and improve logging

### PR DESCRIPTION
**Pull request by alecco**
- To isolate the different pytest suites, remove the top level conftest and move needed contents to existing `test/pylib/cql_repl/conftest.py` and `test/topology/conftest.py`.
- Add logging to CQL and Python suites.
- Log driver version for CQL and topology tests.

### CODE REVIEW

**Comment by alecco** [comment]
This pr is based on top of Kamil's  #11467 (imports fix II) and consists only of 3 commits.

**Comment by kbr-** [comment]
Please rebase

**Comment by alecco** [comment]
Last version rebased on top of master.

File: test/pylib/cql_repl/conftest.py [hunk 1]
```diff
@@ -17,6 +18,13 @@
 from cassandra.cluster import ExecutionProfile     # type: ignore # pylint: disable=no-name-in-module
 from cassandra.cluster import EXEC_PROFILE_DEFAULT  # pylint: disable=no-name-in-module
 from cassandra.policies import RoundRobinPolicy    # type: ignore # pylint: disable=no-name-in-module
+from cassandra.connection import DRIVER_NAME       # type: ignore # pylint: disable=no-name-in-module
+from cassandra.connection import DRIVER_VERSION    # type: ignore # pylint: disable=no-name-in-module
+
+
+logger = logging.getLogger(__name__)
+logger.warning("Driver name %s", DRIVER_NAME)
```
Comment (kbr-) [inline test/pylib/cql_repl/conftest.py:26]: Nit: why warning and not info?

File: test.py [hunk 2]
```diff
@@ -785,6 +785,7 @@ def _reset(self) -> None:
         self.is_after_test_ok = False
         self.args = [
             "-s",  # don't capture print() output inside pytest
+            "--log-level=DEBUG",   # Capture logs
```
Comment (kbr-) [inline test.py:788]: Can you explain which logs exactly this is enabling? Because there are already some logs from all sources (the harness and the test).

**Review by kbr- — APPROVED** [review]
LGTM, I can merge once CI finishes.

**Pull request closed without merging**